### ISSUE TITLE
CI: pkg/kvstore/allocator unit test failed: Timeout while waiting for initial allocator state

### ISSUE THREAD

**Issue by joestringer** [cilium/cilium#24904]
## CI failure

PR #24864 is documenting a change in committers, therefore has no functional changes to the codebase. It failed the integration tests for `pkg/kvstore/allocator`:

https://github.com/cilium/cilium/actions/runs/4698931497/jobs/8331798835?pr=24864

The integration tests do not gather any further artifacts upon failure.

```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃   FAIL  package: github.com/cilium/cilium/pkg/kvstore/allocator   ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛


────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                

level=info msg="Creating etcd client" ConfigPath= KeepAliveHeartbeat=15s KeepAliveTimeout=25s RateLimit=20 subsys=kvstore
level=info msg="Connecting to etcd server..." config= endpoints="[http://127.0.0.1:4002/]" subsys=kvstore
level=info msg="Got lease ID 49ec877f5ecbb43f and the session TTL is 0s" subsys=kvstore
level=info msg="Got lock lease ID 49ec877f5ecbb444" subsys=kvstore
level=info msg="Initial etcd session established" config= endpoints="[http://127.0.0.1:4002/]" subsys=kvstore
level=info msg="Successfully verified version of etcd endpoint" config= endpoints="[http://127.0.0.1:4002/]" etcdEndpoint="http://127.0.0.1:4002/" subsys=kvstore version=3.3.20
level=info msg="Allocated new global key" key=key0001 subsys=allocator
level=info msg="Allocated new global key" key=key0002 subsys=allocator
level=info msg="Allocated new global key" key=key0003 subsys=allocator
level=info msg="Allocated new global key" key=key0004 subsys=allocator
level=info msg="Allocated new global key" key=key0005 subsys=allocator
level=info msg="Allocated new global key" key=key0006 subsys=allocator
level=info msg="Allocated new global key" key=key0007 subsys=allocator
level=info msg="Allocated new global key" key=key0008 subsys=allocator
level=info msg="Allocated new global key" key=key0009 subsys=allocator
level=info msg="Allocated new global key" key=key0010 subsys=allocator
level=info msg="Allocated new global key" key=key0011 subsys=allocator
level=info msg="Allocated new global key" key=key0012 subsys=allocator
level=info msg="Allocated new global key" key=key0013 subsys=allocator
level=info msg="Allocated new global key" key=key0014 subsys=allocator
level=info msg="Allocated new global key" key=key0015 subsys=allocator
level=info msg="Allocated new global key" key=key0016 subsys=allocator
level=info msg="Allocated new global key" key=key0017 subsys=allocator
level=info msg="Allocated new global key" key=key0018 subsys=allocator
level=info msg="Allocated new global key" key=key0019 subsys=allocator
level=info msg="Allocated new global key" key=key0020 subsys=allocator
level=info msg="Allocated new global key" key=key0021 subsys=allocator
level=info msg="Allocated new global key" key=key0022 subsys=allocator
level=info msg="Allocated new global key" key=key0023 subsys=allocator
level=info msg="Allocated new global key" key=key0024 subsys=allocator
level=info msg="Allocated new global key" key=key0025 subsys=allocator
level=info msg="Allocated new global key" key=key0026 subsys=allocator
level=info msg="Allocated new global key" key=key0027 subsys=allocator
level=info msg="Allocated new global key" key=key0028 subsys=allocator
level=info msg="Allocated new global key" key=key0029 subsys=allocator
level=info msg="Allocated new global key" key=key0030 subsys=allocator
level=info msg="Allocated new global key" key=key0031 subsys=allocator
level=info msg="Allocated new global key" key=key0032 subsys=allocator
level=info msg="Reusing existing global key" key=key0001 subsys=allocator
level=info msg="Releasing key" key=key0001 subsys=allocator
level=info msg="Released last local use of key, invoking global release" key=key0001 subsys=kvstorebackend
level=info msg="Reusing existing global key" key=key0002 subsys=allocator
level=info msg="Released last local use of key, invoking global release" key=key0025 subsys=kvstorebackend
level=info msg="Releasing key" key=key0026 subsys=allocator
level=info msg="Released last local use of key, invoking global release" key=key0026 subsys=kvstorebackend
level=info msg="Releasing key" key=key0027 subsys=allocator
level=info msg="Released last local use of key, invoking global release" key=key0027 subsys=kvstorebackend
level=info msg="Releasing key" key=key0028 subsys=allocator
level=info msg="Released last local use of key, invoking global release" key=key0028 subsys=kvstorebackend
level=info msg="Releasing key" key=key0029 subsys=allocator
level=info msg="Released last local use of key, invoking global release" key=key0029 subsys=kvstorebackend
level=info msg="Releasing key" key=key0030 subsys=allocator
level=info msg="Released last local use of key, invoking global release" key=key0030 subsys=kvstorebackend
level=info msg="Releasing key" key=key0031 subsys=allocator
level=info msg="Released last local use of key, invoking global release" key=key0031 subsys=kvstorebackend
level=info msg="Releasing key" key=key0032 subsys=allocator
level=info msg="Released last local use of key, invoking global release" key=key0032 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=23 key=test-prefixRXLHSAiZeLEs/id/23 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=4 key=test-prefixRXLHSAiZeLEs/id/4 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=17 key=test-prefixRXLHSAiZeLEs/id/17 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=2 key=test-prefixRXLHSAiZeLEs/id/2 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=3 key=test-prefixRXLHSAiZeLEs/id/3 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=14 key=test-prefixRXLHSAiZeLEs/id/14 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=16 key=test-prefixRXLHSAiZeLEs/id/16 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=20 key=test-prefixRXLHSAiZeLEs/id/20 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=21 key=test-prefixRXLHSAiZeLEs/id/21 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=27 key=test-prefixRXLHSAiZeLEs/id/27 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=13 key=test-prefixRXLHSAiZeLEs/id/13 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=15 key=test-prefixRXLHSAiZeLEs/id/15 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=18 key=test-prefixRXLHSAiZeLEs/id/18 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=5 key=test-prefixRXLHSAiZeLEs/id/5 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=6 key=test-prefixRXLHSAiZeLEs/id/6 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=1 key=test-prefixRXLHSAiZeLEs/id/1 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=19 key=test-prefixRXLHSAiZeLEs/id/19 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=25 key=test-prefixRXLHSAiZeLEs/id/25 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=26 key=test-prefixRXLHSAiZeLEs/id/26 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=29 key=test-prefixRXLHSAiZeLEs/id/29 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=32 key=test-prefixRXLHSAiZeLEs/id/32 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=9 key=test-prefixRXLHSAiZeLEs/id/9 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=24 key=test-prefixRXLHSAiZeLEs/id/24 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=30 key=test-prefixRXLHSAiZeLEs/id/30 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=7 key=test-prefixRXLHSAiZeLEs/id/7 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=22 key=test-prefixRXLHSAiZeLEs/id/22 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=28 key=test-prefixRXLHSAiZeLEs/id/28 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=8 key=test-prefixRXLHSAiZeLEs/id/8 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=10 key=test-prefixRXLHSAiZeLEs/id/10 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=11 key=test-prefixRXLHSAiZeLEs/id/11 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=12 key=test-prefixRXLHSAiZeLEs/id/12 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=31 key=test-prefixRXLHSAiZeLEs/id/31 subsys=kvstorebackend
level=warning msg="Support for Consul as a kvstore backend has been deprecated due to lack of maintainers. If you are interested in helping to maintain Consul support in Cilium, please reach out on GitHub or the official Cilium slack" subsys=kvstore url="https://cilium.herokuapp.com/"
level=info msg="Allocated new global key" key="MTs=" subsys=allocator
level=info msg="Allocated new global key" key="MTsyOw==" subsys=allocator
level=info msg="Releasing key" key="1;" subsys=allocator
level=info msg="Released last local use of key, invoking global release" key="1;" subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=18 key=test-prefixXeeLHWymBELY/id/18 subsys=kvstorebackend
level=warning msg="Support for Consul as a kvstore backend has been deprecated due to lack of maintainers. If you are interested in helping to maintain Consul support in Cilium, please reach out on GitHub or the official Cilium slack" subsys=kvstore url="https://cilium.herokuapp.com/"
level=info msg="Allocated new global key" key="MTs=" subsys=allocator
level=info msg="Releasing key" key="1;" subsys=allocator
level=info msg="Released last local use of key, invoking global release" key="1;" subsys=kvstorebackend
level=info msg="Allocated new global key" key="Mjs=" subsys=allocator
level=info msg="Releasing key" key="2;" subsys=allocator
level=info msg="Released last local use of key, invoking global release" key="2;" subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=1 key=test-prefixKxEogZFDbshG/id/1 subsys=kvstorebackend
level=fatal msg="Timeout while waiting for initial allocator state" subsys=allocator
```

The underlying error message has been seen before in #23371, #8283, #8385. Possible relation to those failures but that is unknown at this stage.

**Comment by joestringer** [comment]
Assigning to @aanm to test a GH behavior

**Comment by joestringer** [comment]
Gotcha; we can't revert behavior like that.
Assigning to @aanm to fix the bug introduced by #22127.
